### PR TITLE
feat: support custom sampconfig and template files

### PR DESCRIPTION
**Pull request by carlsim0n**
## Description of changes:

- Allow users to specify a custom samconfig file
- Allow users to specify a custom template file
- Updated documentation with the new features
- Refactor how to find and determine runtimes
- Support more samconfig file formats .yml|.yaml

### CODE REVIEW

**Comment by carlsim0n** [comment]
Need help with verifying:
- CDK with TS
- dotnet
- python

File: README.md [hunk 1]
```diff
@@ -119,6 +127,8 @@ Options:
   -d, --debug                   Configure debug for vscode. This only needs to be run once per project (default: false)
   -p, --profile [profile]       AWS profile to use
   --region [region]             The AWS region to use. Falls back on AWS_REGION environment variable if not specified
+  --template-file               If you wish to provide a custom named template file. Otherwise samp will look for template.|yaml|yml|json
+  --samconfig-file              If you wish to provide a custom samconfig file. Default is samconfig.toml|yaml|yml|json
```
Comment (ljacobsson) [inline README.md:131]: JSON isn't supported by SAM as a format for samconfig, so please remove that option

File: src/commands/local/index.js [hunk 2]
```diff
@@ -13,6 +13,8 @@ program
   .option("-d, --debug", "Configure debug for vscode. This only needs to be run once per project", false)
   .option("-p, --profile [profile]", "AWS profile to use")
   .option("--region [region]", "The AWS region to use. Falls back on AWS_REGION environment variable if not specified")
+  .option("--template-file [templateFile]", "If you wish to provide a custom named template file. Otherwise samp will look for template.|yaml|yml|json")
+  .option("--samconfig-file [samconfigFile]", "If you wish to provide a custom samconfig file. Default is samconfig.toml|yaml|yml|json")
```
Comment (ljacobsson) [inline src/commands/local/index.js:17]: No JSON

**Review by ljacobsson — COMMENTED** [review]
Will test a bit on my machine before approving

File: README.md (hunk 1 above)
Comment (carlsim0n) [reply README.md]: fixed!

File: src/commands/local/index.js (hunk 2 above)
Comment (carlsim0n) [reply src/commands/local/index.js]: fixed!

**Pull request closed without merging**